library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitlePainter.java

```diff
@@ -355,6 +355,22 @@ private void setupTextLayout() {
   }
 
   private void setupBitmapLayout() {
+    // Default position
+    if (cuePosition == Cue.DIMEN_UNSET) {
+      cuePosition = 0.5f;
+    }
+
+    // Default line
+    if (cueLine == Cue.DIMEN_UNSET) {
+      cueLineAnchor = Cue.ANCHOR_TYPE_MIDDLE;
+      cueLine = 0.85f;
+    }
+
+    // Default width
+    if (cueSize == Cue.DIMEN_UNSET) {
+      cueSize = 0.5f;
```

Comment:
The way these defaults work can cause a subtitle to be drawn in quite a weird way, I think. This looks like it'll default the cue to occupy 50% of the screen horizontally. Suppose the bitmap subtitle is a very short word, like "Hi". Wont it end up being sized to occupy nearly the entire vertical space in this case :)?

You may need to do something a little more clever, where you define a box within which you want the bitmap to be displayed (e.g. 50% of space horizontally and 10% vertically), and then figure out how to compute the `bitmapRect` so that it doesn't go outside of that box.

Reply:
You are right :)

And lets say "Hi" and "Hi, how are you?" should have the same height, that's not case now. I'll try to find a solutions for this today.

The other adjustments based on the comments are done, but haven't pushed those yet.

Reply:
In the general case I don't think it's possible for you to keep the line height the same if the region hasn't been defined. Doing so would require you to know how many lines are in the bitmap, and you don't know that (so really, having the region being defined properly is fairly important). I think fitting withing a box that constraints it to a sensibly small portion of the screen is the best you can do.

Reply:
One idea is to fix the height at 10% as you said and make the bitmap to fit this height (keeping aspect ratio), however if the bitmap goes beyond the box's width (50%) then will be scaled down by its width until hits the "box" (keeping aspect ratio here as well). I think this is an acceptable solution for "one line" images, however if the cue has two lines then the "line height" will be the half of specified box's height (= 5%). I don't see a solution for this particular case, because we don't know all of the bitmap's sizes ahead (e.g. live tv).

Do you think this is a viable solution?

Reply:
I think what you describe is the best you can do (short of something crazy like using OCR to figure out how many lines there are - which isn't something we're realistically going to do :)).

Reply:
Probably it's easier to add support for regions that are defined by pixels, than using OCR, however nice idea, hehe :)

Thanks for the quick response, I will do the way I described then.